### PR TITLE
Bugfix/objects 997 url encoding

### PR DESCRIPTION
**Pull request by andreassiegel**
### What changes were proposed in this pull request?
- moves the `RequestEntity` creation out of the `RestConnector` classes into their own utility components
- applies URL encoding
- prepares the `RestConnectors` for removal in the near future
### How is this patch documented?

Code, Javadoc.
### How was this patch tested?

Existing and additional unit tests.
#### Depends On

Nothing.

### CODE REVIEW

**Comment by andreassiegelrfid** [comment]
I reverted the package changes and those changes where nothing else than the code style changed.

These changes will happen again in another pull request, but it cleans up the diff for this one.

File: src/main/java/net/smartcosmos/edge/things/rest/template/thing/ThingRestConnector.java [hunk 1]
```diff
@@ -3,153 +3,71 @@
...
 import net.smartcosmos.edge.things.domain.local.things.RestThingCreateResponseDto;
 import net.smartcosmos.edge.things.domain.local.things.RestThingResponse;
 import net.smartcosmos.edge.things.domain.local.things.RestThingUpdate;
 import net.smartcosmos.edge.things.rest.RestTemplateFactory;
 import net.smartcosmos.edge.things.rest.template.SmartCosmosRequest;
+import net.smartcosmos.edge.things.rest.request.ThingRequestFactory;
 
 @Component
 @Slf4j
+@Deprecated
```
Comment (jbauerrfid) [inline src/main/java/net/smartcosmos/edge/things/rest/template/thing/ThingRestConnector.java]: Deprecation of internally used class:
Please add JIRA ticket to update usages of this class to epic https://smartractechnology.atlassian.net/browse/OBJECTS-851

File: src/main/java/net/smartcosmos/edge/things/rest/template/metadata/MetadataRestConnector.java [hunk 2]
```diff
@@ -5,39 +5,34 @@
...
-import net.smartcosmos.edge.things.config.SmartCosmosEdgeThingsProperties;
 import net.smartcosmos.edge.things.domain.local.metadata.RestMetadataCreateResponseDto;
-import net.smartcosmos.edge.things.domain.local.things.RestThingCreate;
 import net.smartcosmos.edge.things.rest.RestTemplateFactory;
-import net.smartcosmos.edge.things.rest.template.SmartCosmosRequest;
+import net.smartcosmos.edge.things.rest.request.MetadataRequestFactory;
 
 @Component
 @Slf4j
+@Deprecated
```
Comment (jbauerrfid) [inline src/main/java/net/smartcosmos/edge/things/rest/template/metadata/MetadataRestConnector.java]: Deprecation of internally used class:
Please add JIRA ticket to update usages of this class to epic https://smartractechnology.atlassian.net/browse/OBJECTS-851

File: src/main/java/net/smartcosmos/edge/things/utility/UrlEncodingUtil.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     *
+     * @param value the input String
+     * @return the encoded value
+     */
+    public static String encode(String value) {
+
+        try {
+            return UriUtils.encode(value, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            log.error("UTF-8 encoding is not supported");
```
Comment (jbauerrfid) [inline src/main/java/net/smartcosmos/edge/things/utility/UrlEncodingUtil.java]: When the function continues to operate I would rather expect a `log.warn()`, not a `log.error()`.

File: src/main/java/net/smartcosmos/edge/things/rest/template/metadata/MetadataRestConnector.java (hunk 2 above)
Comment (andreassiegelrfid) [inline src/main/java/net/smartcosmos/edge/things/rest/template/metadata/MetadataRestConnector.java]: https://smartractechnology.atlassian.net/browse/OBJECTS-1006

**Comment by jbauerrfid** [comment]
Looks good now.

**Pull request merged**